### PR TITLE
feat(testing): add e2e testing framework

### PR DESCRIPTION
**Pull request by nanjingfm**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [`spec` PR link](https://github.com/katanomi/spec) included
- [x] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [x] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [x] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: testing/framework/config.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+func (c *configCondition) Condition(testCtx *TestContext) error {
+	configData, err := NewE2EConfig(c.name).GetConfig(testCtx.Context, testCtx.Client)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal([]byte(configData), c.obj); err != nil {
+		return err
+	}
+
+	testCtx.Context = context.WithValue(testCtx.Context, configCondition{}, c.obj)
```
Comment (nanjingfm) [inline testing/framework/config.go:55]: 将插件的配置塞到context中，后面有类似自定义的全局性的东西（配置、client）都可以塞到context中。

File: testing/framework/config.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	lock sync.RWMutex
+	once sync.Once
+
+	data string
+}
+
+func (c *TestConfig) initData(ctx context.Context, clt client.Client) error {
+	var err error
+	c.once.Do(func() {
```
Comment (nanjingfm) [inline testing/framework/config.go:81]: 通常用于测试初始化阶段，配置只取一次。

File: testing/framework/context.go [hunk 3]
```diff
@@ -18,21 +18,44 @@ package framework
...
 	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // TestContext a test context
 type TestContext struct {
 	Context context.Context
 	Config  *rest.Config
-	Opts    Options
+	Client  client.Client
```
Comment (nanjingfm) [inline testing/framework/context.go:34]: k8s client，因为用的比较多，所以就单独加了个字段（而没有塞到context中）

File: testing/framework/config.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+const (
+	e2eConfigNSKey   = "E2E_CONFIG_NAMESPACE"
+	e2eConfigNameKey = "E2E_CONFIG_NAME_PREFIX"
+)
+
+var (
+	e2eConfigNs         = GetDefaultEnv(e2eConfigNSKey, "katanomi-e2e")
+	e2eConfigNamePrefix = GetDefaultEnv(e2eConfigNameKey, "e2e-config")
```
Comment (danielfbm) [inline testing/framework/config.go:21]: 这些环境变量需要在readme里面提供说明？

File: testing/framework/env_test.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package framework
```
Comment (danielfbm) [inline testing/framework/env_test.go:1]: test文件也需要copyright license信息？

**Review by danielfbm — CHANGES_REQUESTED** [review]

**Review by danielfbm — APPROVED** [review]

**Pull request merged**